pkg/cmd/pulumi/new.go

```diff
@@ -321,11 +321,15 @@ func runNew(ctx context.Context, args newArgs) error {
 	proj.Name = tokens.PackageName(args.name)
 	proj.Description = &args.description
 	proj.Template = nil
+	// TODO[pulumi/pulumi/issues/16309]: This should be handled by the language runtime.
 	// Workaround for python, most of our templates don't specify a venv but we want to use one
 	if proj.Runtime.Name() == "python" {
-		// If the template does give virtualenv use it, else default to "venv"
-		if _, has := proj.Runtime.Options()["virtualenv"]; !has {
-			proj.Runtime.SetOption("virtualenv", "venv")
+		// If we are not using poetry, backfill the virtualenv option.
+		tc, hasToolchain := proj.Runtime.Options()["toolchain"]
+		if !hasToolchain || tc != "poetry" {
```

Comment:
Feel like this should check for the toolchains which would support venv, rather than checking for not poetry?

Reply:
Definitely, updated